python/iceberg/hive/hive_tables.py

```diff
@@ -34,12 +42,59 @@ def __init__(self, conf):
     def new_table_ops(self, conf, database, table):
         return HiveTableOperations(conf, self.get_client(), database, table)
 
-    def drop(self, database, table):
-        raise RuntimeError("Not yet implemented")
+    def drop(self, database, table, purge=False) -> None:
+        ops = self.new_table_ops(self.conf, database, table)
+        metadata = ops.current()
 
-    def get_client(self):
+        with self.get_client() as open_client:
+            _logger.info("Deleting {database}.{table} from Hive Metastore".format(database=database, table=table))
+            open_client.drop_table(database, table, deleteData=False)
+
+        if purge:
+            if metadata is not None:
```

Comment:
I had something a bit simpler in mind. I think we can avoid some of the itertools stuff by just deleting as we find files. This also implies that the `_delete_file` method catches deletes that happen twice

``` python
            if metadata is not None:
                with Pool(self.conf.get(WORKER_THREAD_POOL_SIZE_PROP,
                          cpu_count())) as delete_pool:
                    manifest_lists_to_delete = [ops.current_metadata_location]

                    for s in metadata.snapshots:
                        delete_pool.map(self._delete_file(ops), (i.path() for i in s.get_filtered_manifests().iterator()))
                        delete_pool.map(self._delete_file(ops), s.manifests)
                        if s.manifest_location is not None:
                            manifest_lists_to_delete.append(s.manifest_location)
                    delete_pool.map(self._delete_file(ops), manifest_lists_to_delete)
```

I didn't test this code so it may not be quite right. And some manifests mentioned in later snapshots may already be deleted so can't be read (but can then just be skipped).